app/http.php

```diff
@@ -0,0 +1,1039 @@
+        case Storage::DEVICE_LINODE:
+            $linodeAccessKey = (string) App::getEnv('OPEN_RUNTIMES_EXECUTOR_STORAGE_LINODE_ACCESS_KEY', '');
+            $linodeSecretKey = (string) App::getEnv('OPEN_RUNTIMES_EXECUTOR_STORAGE_LINODE_SECRET', '');
+            $linodeRegion = (string) App::getEnv('OPEN_RUNTIMES_EXECUTOR_STORAGE_LINODE_REGION', '');
+            $linodeBucket = (string) App::getEnv('OPEN_RUNTIMES_EXECUTOR_STORAGE_LINODE_BUCKET', '');
+            $linodeAcl = 'private';
+            return new Linode($root, $linodeAccessKey, $linodeSecretKey, $linodeBucket, $linodeRegion, $linodeAcl);
+        case Storage::DEVICE_WASABI:
+            $wasabiAccessKey = (string) App::getEnv('OPEN_RUNTIMES_EXECUTOR_STORAGE_WASABI_ACCESS_KEY', '');
+            $wasabiSecretKey = (string) App::getEnv('OPEN_RUNTIMES_EXECUTOR_STORAGE_WASABI_SECRET', '');
+            $wasabiRegion = (string) App::getEnv('OPEN_RUNTIMES_EXECUTOR_STORAGE_WASABI_REGION', '');
+            $wasabiBucket = (string) App::getEnv('OPEN_RUNTIMES_EXECUTOR_STORAGE_WASABI_BUCKET', '');
+            $wasabiAcl = 'private';
+            return new Wasabi($root, $wasabiAccessKey, $wasabiSecretKey, $wasabiBucket, $wasabiRegion, $wasabiAcl);
+    }
+}
+
+function removeAllRuntimes(Pool $orchestrationPool): void
+{
+    Console::log('Cleaning up containers before shutdown...');
+
+    $connection = $orchestrationPool->pop();
+    $orchestration = $connection->getResource();
+    $functionsToRemove = $orchestration->list(['label' => 'openruntimes-executor=' . System::getHostname()]);
+    $orchestrationPool->push($connection);
+
+    if (\count($functionsToRemove) === 0) {
+        Console::info('No containers found to clean up.');
+    }
+
+    $callables = [];
+
+    foreach ($functionsToRemove as $container) {
+        $callables[] = function () use ($container, $orchestrationPool) {
+            try {
+                $connection = $orchestrationPool->pop();
+                $orchestration = $connection->getResource();
+                $orchestration->remove($container->getId(), true);
+                Console::success('Removed container ' . $container->getName());
+            } catch (\Throwable $th) {
+                Console::error('Failed to remove container: ' . $container->getName());
+                Console::error($th);
+            } finally {
+                isset($connection) && $orchestrationPool->push($connection);
+            }
+        };
+    }
+
+    /** @phpstan-ignore-next-line */
+    Swoole\Coroutine\batch(
+        $callables
+    );
+
+    Console::success('Done.');
+}
+
+App::post('/v1/runtimes')
+    ->desc("Create a new runtime server")
+    ->param('runtimeId', '', new Text(64), 'Unique runtime ID.')
+    ->param('image', '', new Text(128), 'Base image name of the runtime.')
+    ->param('source', '', new Text(0), 'Path to source files.')
```

Comment:
Might make more sense to have source and destination grouped one after the other.

Reply:
Moved together.